plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobContainer.java

```diff
@@ -106,12 +105,18 @@ public DeleteResult delete() throws IOException {
     }
 
     @Override
-    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
-        List<String> blobsWithFullPath = blobNames.stream()
-            .map(this::buildKey)
```

Comment:
This was a pretty nasty pattern that we used in a bunch of spots, basically doubling the peak memory use for a delete (which is far from ideal when deleting a million blobs or so here).

Reply:
agreed

Reply:
+1